### ISSUE TITLE
unable to attach the target

### ISSUE THREAD

**Issue by qgy123** [chame1eon/jnitrace#3]
Device: Google Nexus 5
Frida-server:frida-server-12.6.13-android-arm
frida:12.6.13
frida-tools:2.0.2
Application: just a helloworld

Stack Trace:
```
C:\Users\YueLuo>jnitrace -l * com.yueluo.bkpttest -m attach
Traceback (most recent call last):
  File "c:\users\yueluo\appdata\local\programs\python\python37-32\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\users\yueluo\appdata\local\programs\python\python37-32\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\YueLuo\AppData\Local\Programs\Python\Python37-32\Scripts\jnitrace.exe\__main__.py", line 9, in <module>
  File "c:\users\yueluo\appdata\local\programs\python\python37-32\lib\site-packages\jnitrace\jnitrace.py", line 491, in main
    session = device.attach(device.attach(args.target))
  File "c:\users\yueluo\appdata\local\programs\python\python37-32\lib\site-packages\frida\core.py", line 110, in attach
    return Session(self._impl.attach(self._pid_of(target)))
  File "c:\users\yueluo\appdata\local\programs\python\python37-32\lib\site-packages\frida\core.py", line 128, in _pid_of
    return self.get_process(target).pid
  File "c:\users\yueluo\appdata\local\programs\python\python37-32\lib\site-packages\frida\core.py", line 68, in get_process
    process_name_lc = process_name.lower()
AttributeError: 'Session' object has no attribute 'lower'
```

I have fixed this problem by simply change
jni.py
#487 to
```py
    device = frida.get_usb_device(3)
    pid = device.spawn([args.target])
    
    if args.inject_method == "spawn":
        session = device.attach(pid)
    else:
        session = device.attach(args.target)
```

the issuse seems like the lastest frida has changed their api, but I know shit about frida, so I'm not sure how to completely fix this problem on all frida versions.

**Comment by chame1eon** [comment]
Hi, thanks very much for the bug report. Good spot. The issue was with my code and not Frida. I have fixed it and pushed a new version to pip, version 2.0.1.

Please can you try it and let me know if the problem is solved?

One last thing, I would recommend using the default attach option (-m spawn), rather than -m attach. Using attach may miss some uses of the JNI API on start up.

Thanks 😃

**Comment by qgy123** [comment]
Hi, I upgraded the script, it works fine!
I'm using attach options because I only care about a function which was suspicious.
Thank you for your work! It's flexible and better than I add some dirty code in android source code.

**Comment by chame1eon** [comment]
Hi, just to let you know I have added the -I and -E options to allow you to include or exclude library exports from the trace. It should help you with filtering just the function you care about :).

You can run it as follows:

jnitrace -I Java_com_nativetest_MainActivity_stringFromJNI -l libnative-lib.so com.example.myapplication

Available through pip as version 2.1.0
